analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -102,135 +88,60 @@ protected override void Initialize(SonarAnalysisContext context)
-                        return variableSyntax.Identifier.GetLocation();
-                    }
-
-                    var eventSyntax = reference.GetSyntax() as EventDeclarationSyntax;
-                    return eventSyntax?.Identifier.GetLocation();
-                },
-                "event");
-        }
-
-        private static void CheckField(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IFieldSymbol field)
+        private static void CheckFieldOrProperty<T>(T propertyOrField,
+                                                    INamedTypeSymbol containerClassSymbol,
+                                                    SymbolAnalysisContext context,
+                                                    string memberType) where T : ISymbol
         {
-            CheckFieldOrProperty(field, containerClassSymbol, context,
-                f =>
-                {
-                    var reference = f.DeclaringSyntaxReferences.FirstOrDefault();
-                    var syntax = reference?.GetSyntax() as VariableDeclaratorSyntax;
-                    return syntax?.Identifier.GetLocation();
-                },
-                "field");
-        }
-
-        private static void CheckProperty(SymbolAnalysisContext context, INamedTypeSymbol containerClassSymbol, IPropertySymbol property)
-        {
-            CheckFieldOrProperty(property, containerClassSymbol, context, p =>
-            {
-                var reference = p.DeclaringSyntaxReferences.FirstOrDefault();
-                if (reference == null)
-                {
-                    return null;
-                }
-                if (!(reference.GetSyntax() is PropertyDeclarationSyntax syntax))
-                {
-                    return null;
-                }
-                return syntax.Identifier.GetLocation();
-            }, "property");
-        }
-
-        private static void CheckFieldOrProperty<T>(T propertyOrField, INamedTypeSymbol containerClassSymbol,
-            SymbolAnalysisContext context, Func<T, Location> locationSelector, string memberType) where T : ISymbol
-        {
-            var shadowsProperty = GetSelfAndOuterClasses(containerClassSymbol)
+            var selfAndOutterClasses = GetSelfAndOuterClasses(containerClassSymbol);
+            var shadowsProperty = selfAndOutterClasses
                 .SelectMany(c => c.GetMembers(propertyOrField.Name))
                 .OfType<IPropertySymbol>()
                 .Any(prop => prop.IsStatic);
-            var shadowsField = GetSelfAndOuterClasses(containerClassSymbol)
+            var shadowsField = selfAndOutterClasses
                 .SelectMany(c => c.GetMembers(propertyOrField.Name))
                 .OfType<IFieldSymbol>()
                 .Any(field => field.IsStatic || field.IsConst);
 
-            if (shadowsProperty || shadowsField)
+            if ((shadowsProperty || shadowsField) && propertyOrField.GetFirstIdentifier()?.GetLocation() is { } location)
```

Comment:
Let's exract only the `identifier` here, as the `GetLocation()` will never return `null` for a known token. Current version implies that it can happen.
```suggestion
            if ((shadowsProperty || shadowsField) && propertyOrField.GetFirstIdentifier() is { } identifier)
```

Reply:
No, we should discover that once it is in the language as part of the specification sprint for new language features.